docs/hosting/configuration/task-runners.md

```diff
@@ -0,0 +1,83 @@
+
+![Task runner overview](/_images/hosting/configuration/task-runner-concept.png)
+
+Task runners connect to the task broker using a websocket connection. A task requester submits a task request to the broker where an available task runner can pick it up for execution.
+
+The runner executes the task and submits the results to the task requester. The task broker coordinates communication between the runner and the requester.
+
+The n8n instance (main and worker) acts as the broker. The Code node in this case is the task requester.
+
+## Task runner modes
+
+You can use task runners in two different modes: internal and external.
+
+### Internal mode
+
+In internal mode, the n8n instance launches the task runner as a child process. The n8n process monitors and manages the life cycle of the task runner. The task runner process shares the same `uid` and `gid` as n8n.
+
+### External mode
+
+In external mode, an external orchestrator (for example, Kubernetes) launches the task runner instead of n8n. Typically, this means you would configure the task runner to run as a side-car container next to n8n.
+
+![Task runner deployed as a side-car container](/_images/hosting/configuration/task-runner-external-mode.png)
+
+In this mode, the orchestrator monitors and manages the life cycle of the task runner container. The task runner is fully isolated from the n8n instance.
+
+When using the [Queue mode](/hosting/scaling/queue-mode), each n8n container (main and workers) needs to have its own task runner.
+
+## Setting up external mode
+
+Use the following details to configure task runners in external mode
+
+### Configuring n8n instance in external mode
+
+You can configure n8n to use external task runners by setting the following environment variables:
+
+| Environment variables                                  | Description                                                |
+|--------------------------------------------------------|------------------------------------------------------------|
+| `N8N_RUNNERS_ENABLED=true`                             | Enables task runners.                                      |
+| `N8N_RUNNERS_MODE=external`                            | Use task runners in external mode.                         |
+| `N8N_RUNNERS_AUTH_TOKEN=<random secure shared secret>` | A shared secret task runners use to connect to the broker. |
+
+For full list of environment variables see [task runner environment variables](/hosting/configuration/environment-variables/task-runners/).
+
+### Configuring task runners in external mode
+
+The task runner comes bundled within the n8n Docker image. The Docker image also includes the task runner launcher.
+
+The launcher can start the runner on-demand, which means lower memory usage when there's no work needed, but a short delay (few hundred ms) in cold-start. The launcher also monitors the launcher and restarts it in case of infinite loops or other issues.
```

Comment:
```suggestion
The launcher can start the runner on-demand, which means lower memory usage when there's no work needed, but a short delay (few hundred ms) in cold-start. The launcher also monitors the runner and restarts it in case of infinite loops or other issues.
```